src/windows/wslaservice/exe/WSLASession.cpp

```diff
@@ -13,15 +13,27 @@ Module Name:
 --*/
 
 #include "WSLASession.h"
+#include "WSLAUserSession.h"
 
 using wsl::windows::service::wsla::WSLASession;
 
-WSLASession::WSLASession(const WSLA_SESSION_SETTINGS& Settings) : m_displayName(Settings.DisplayName)
+WSLASession::WSLASession(const WSLA_SESSION_SETTINGS& Settings, WSLAUserSessionImpl& userSessionImpl, const VIRTUAL_MACHINE_SETTINGS& VmSettings) :
+    m_sessionSettings(Settings),
```

Comment:
nit: maybe for a future change. We probably shouldn't keep that structure as a field since some of its pointers will be invalid when the constructor returns. (and since it's not used in other places)

Reply:
I did reply about this in an earlier comment. Having the struct here will need a revisit.

Reply:
Added a TODO around this as well